core/src/avm1/scope.rs

```diff
@@ -30,10 +29,10 @@ pub enum ScopeClass {
 }
 
 /// Represents a scope chain for an AVM1 activation.
-#[derive(Debug, Collect)]
+#[derive(Copy, Clone, Debug, Collect)]
```

Comment:
I prefer keeping large structs non-`Copy`, since cloning them isn't cheap.

Reply:
How large is 'large', here? We have two GC pointers - one optional, and one enumclass'd; and a binary enum. I doubt it exceeds 32 bytes.

Reply:
With alignment, it's 24 bytes on x86-64, and 12 bytes on WASM, so definitely not a "big" struct.

Reply:
Actually I consider structs larger than a pointer size as "big" enough. In my opinion implicit clones for big structs hurt code readability (See `new_target_scope` for example, it's very unclear that `scope` is a clone of `parent`, at least for me).

Reply:
Gotcha. So the rule is `size_of::<T>() > size_of::<usize>()` should not be `Copy`.

I wonder if this could be a custom lint...

Reply:
Well, by this rule `&[T]` shouldn't be `Copy` :D 
Will remove the Copy impl anyways, I don't think it's actually used much.